transport/src/test/java/io/netty/channel/ChannelInitializerTest.java

```diff
@@ -0,0 +1,105 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.local.LocalAddress;
+import io.netty.channel.local.LocalChannel;
+import io.netty.channel.local.LocalServerChannel;
+import org.junit.After;
+
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.concurrent.TimeUnit;
+
+public class ChannelInitializerTest {
+
+    private static final int TIMEOUT_MILLIS = 1000;
+    private static final LocalAddress SERVER_ADDRESS = new LocalAddress("addr");
+    private EventLoopGroup group;
+    private ServerBootstrap server;
+    private Bootstrap client;
+
+    @Before
+    public void setUp() {
+        group = new DefaultEventLoopGroup(1);
+        server = new ServerBootstrap()
+                .group(group)
+                .channel(LocalServerChannel.class)
+                .localAddress(SERVER_ADDRESS);
+        client = new Bootstrap()
+                .group(group)
+                .channel(LocalChannel.class)
+                .handler(new ChannelInitializer<LocalChannel>() {
+                    @Override
+                    public void initChannel(LocalChannel channel) {
+                    }
+                });
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        group.shutdownGracefully(0, TIMEOUT_MILLIS, TimeUnit.MILLISECONDS).sync();
+    }
+
+    @Test(timeout = TIMEOUT_MILLIS)
+    public void firstHandlerInPipelineShouldReceiveChannelRegisteredEvent() throws Exception {
+        final InspectableHandler testSubject = new InspectableHandler();
+        server.childHandler(new ChannelInitializer<LocalChannel>() {
+            @Override
+            public void initChannel(LocalChannel channel) {
+                channel.pipeline().addFirst(testSubject);
+            }
+        });
+        server.bind().sync();
+        client.connect(SERVER_ADDRESS).sync();
+        assertTrue(testSubject.channelRegisteredFired);
+    }
+
+    @Test(timeout = TIMEOUT_MILLIS)
+    public void lastHandlerInPipelineShouldReceiveChannelRegisteredEvent() throws Exception {
+        final InspectableHandler testSubject = new InspectableHandler();
+        server.childHandler(new ChannelInitializer<LocalChannel>() {
+            @Override
+            public void initChannel(LocalChannel channel) {
+                channel.pipeline().addLast(testSubject);
+            }
+        });
+        server.bind().sync();
+        client.connect(SERVER_ADDRESS).sync();
+        assertTrue(testSubject.channelRegisteredFired);
+    }
```

Comment:
Also you can share most of the test-code and just use a different ChannelInitializer for each of them.